src/util/pg.py

```diff
@@ -196,27 +196,58 @@ def explode_query_range(cr, query, table, alias=None, bucket_size=10000, prefix=
 
     alias = alias or table
 
-    cr.execute("SELECT min(id), max(id) FROM {}".format(table))
+    if "{parallel_filter}" not in query:
+        sep_kw = " AND " if re.search(r"\sWHERE\s", query, re.M | re.I) else " WHERE "
+        query += sep_kw + "{parallel_filter}"
+
+    cr.execute(format_query(cr, "SELECT min(id), max(id) FROM {}", table))
     min_id, max_id = cr.fetchone()
     if min_id is None:
         return []  # empty table
+    count = (max_id + 1 - min_id) // bucket_size
+    if count > 100000:
```

Comment:
I find this threshold to be problematic. For a small table with an off sequence space this may be good, for bigger tables, it is unclear why this threshold is reached. How about determining the proportion of sequence space (max - min) to the  live tuple count? Something like
```suggestion
    cr.execute(format_query(cr, "SELECT min(id), max(id), count(id) FROM {}", table))
    min_id, max_id, count_id = cr.fetchone()
    if min_id is None:
        return []  # empty table
    if (max_id + 1 - min_id) / count_id > 10:
```

Reply:
We discussed this in DM (with @KangOl)

Empty buckets are fine as long as they run fast. If we do as you propose we would be switching to the slower strategy for queries that otherwise will just run fast. OTOH the _real_ issue here is the MemoryError. Thus average-//-query-size * 100K looks acceptable as a threshold. We can adapt the value in the future if it proves problematic.

The precise bucketing could be slow due to (already tested) high `Heap fetches`. In the end there is no good option for all cases.